lib/simulator.js

```diff
@@ -36,27 +37,37 @@ async function getSimulator (udid) {
     throw new Error(`No sim found with udid ${udid}`);
   }
 
-  handleUnsupportedXcode(xcodeVersion);
-
   log.info(`Constructing iOS simulator for Xcode version ${xcodeVersion.versionString} ` +
            `with udid '${udid}'`);
-  if (xcodeVersion.major === 6) {
-    return new SimulatorXcode6(udid, xcodeVersion);
-  } else if (xcodeVersion.major >= 7 && xcodeVersion.major < 8) {
-    if (xcodeVersion.minor < 3) {
-      return new SimulatorXcode7(udid, xcodeVersion);
-    } else {
-      return new SimulatorXcode73(udid, xcodeVersion);
-    }
-  } else if (xcodeVersion.major === 8) {
-    return new SimulatorXcode8(udid, xcodeVersion);
-  } else if (xcodeVersion.major === 9) {
-    if (xcodeVersion.minor < 3) {
-      return new SimulatorXcode9(udid, xcodeVersion);
-    } else {
-      return new SimulatorXcode93(udid, xcodeVersion);
-    }
+  let SimClass;
+  switch (xcodeVersion.major) {
+    case 6:
+      SimClass = SimulatorXcode6;
+      break;
+    case 7:
+      if (xcodeVersion.minor < 3) {
+        SimClass = SimulatorXcode7;
+      } else {
+        SimClass = SimulatorXcode73;
+      }
+      break;
+    case 8:
+      SimClass = SimulatorXcode8;
+      break;
+    case 9:
+      if (xcodeVersion.minor < 3) {
+        SimClass = SimulatorXcode9;
+      } else {
+        SimClass = SimulatorXcode93;
+      }
+      break;
+    case 10:
+      SimClass = SimulatorXcode10;
+      break;
+    default:
+      handleUnsupportedXcode(xcodeVersion);
```

Comment:
Cannot we just default to the most recent simulator class by default instead of throwing an exception? I don't see any need in having derived classes, that are empty